adala/runtimes/_litellm.py

```diff
@@ -283,12 +303,33 @@ async def batch_to_batch(
         # convert list of LLMResponse objects to the dataframe records
         df_data = []
         for response in responses:
-            if isinstance(response, Exception):
-                error_message = type(response).__name__
-                # error_details = str(response)
-                error_details = traceback.format_exc()
+            if isinstance(response, InstructorRetryException):
+                e = response
+                # get root cause error from retries
+                n_attempts = e.n_attempts
+                e = e.__cause__.last_attempt.exception()
```

Comment:
On the fence about this, but it is definitely tempting to just make a function to convert to `error_dct` from `e` at least... the logging can still be determined in each section for itself imo, but seems like the sort of conversion we may end up with in a couple places?

Reply:
Like an `instructor_retry_exception_repr` or `format_instructor_retry_exception` kind of thing. Definitely not necessary though, whatever seems right to you

Reply:
yep, did this. Kept the branch on error type outside the function for clarity